sdk/src/main/java/io/customer/sdk/CustomerIO.kt

```diff
@@ -52,6 +54,16 @@ class CustomerIO internal constructor(
             return this
         }
 
+        /**
+         * Override url/deep link handling
+         *
+         * @param urlHandler callback called when deeplink push action is performed.
+         */
+        fun setCustomerIOUrlHandler(urlHandler: CustomerIOUrlHandler): Builder {
+            this.urlHandler = urlHandler
+            return this
+        }
+
```

Comment:
What use case would having a url handler help with? I have wondered about a url handler as well, but wasn't sure if it provided enough value because [apps can setup deep link handling](https://developer.android.com/training/app-links/deep-linking) in their app. My thought is that [setting up deep link handling](https://developer.android.com/training/app-links/deep-linking) is a universal way for an app to handle deep links not dependent on an SDK or tool.

Reply:
This is there for the applications that do the internal Deep linking managements via some Deep linking manager. For [example](https://proandroiddev.com/an-effective-deep-link-navigation-approach-for-multi-modular-android-applications-63860a3f45b3), applications that use multi modules approach and activities in each module. Or an activity, that has complex navigation graphs and supports fragment navigations of deep link internally, or any application that uses 3rd party deep linking [library](https://github.com/airbnb/DeepLinkDispatch). 

So we support both cases, android recommended `Manifest` way or URL handler.